### DISCUSSION TITLE
How to replace selected text?

### DISCUSSION THREAD

**Discussion by rogerl33** [danielsaidi/RichTextKit#197] [Q&A] [answered]
If I do a SelectRange, and then a Paste, RichTextKit only does an insertion not a replacement. Is there some other way to achieve what I want?
Thanks!

**Comment by danielsaidi** [comment]
This must be an oversight from my side. Pasting should behave as you describe.

**Comment by chnbr** [comment]
Im having problems with the following line:

context.handle(.replaceText(in: myRange, with: attrText))

It seems that attrText is not replacing myRange, but is inserted at .location of myRange. 

Can you confirm this (and if so correct it) ?

Thanks

**Reply by danielsaidi** [reply]
Yeah, replace should take the length of the provided range. I will check.

**Comment by chnbr** [comment]
I have debugged into and it seems that the range argument is not passed correctly. I pass a non-null range, but in RichTextCoordinator+Actions.swift I am seeing NSRange(0,0). That would explain the behaviour. However, I was not able to find where the range gets lost...

**Comment by chnbr** [comment]
My update is definitely running on @mainactor... I do see immediate updates, but they replace the 0,0 range all the time.
Everything is "replaced" at location 0.

**Comment by chnbr** [comment, marked as answer]
I am very sorry. My fault. I passed an old value of my range. Everything is OK !!!

So, sorry ! Apologies !

**Comment by danielsaidi** [comment]
Haha ok, so can we close this?

**Comment by chnbr** [comment]
Yes 🙏
